library/azure_rm_rediscache.py

```diff
@@ -0,0 +1,618 @@
+            ),
+            location=dict(
+                type='str'
+            ),
+            sku=dict(
+                type='dict',
+                options=sku_spec
+            ),
+            enable_non_ssl_port=dict(
+                type='bool',
+                default=False
+            ),
+            maxfragmentationmemory_reserved=dict(
+                type='int'
+            ),
+            maxmemory_reserved=dict(
+                type='int'
+            ),
+            maxmemory_policy=dict(
+                type='str',
+                choices=[
+                    "volatile_lru",
+                    "allkeys_lru",
+                    "volatile_random",
+                    "allkeys_random",
+                    "volatile_ttl",
+                    "noeviction"
+                ]
+            ),
+            notify_keyspace_events=dict(
+                type='int'
+            ),
+            shard_count=dict(
+                type='ints'
+            ),
+            static_ip=dict(
+                type='str'
+            ),
+            subnet=dict(
+                type='raw'
+            ),
+            tenant_settings=dict(
+                type='dict'
+            ),
+            state=dict(
+                type='str',
+                default='present',
+                choices=['present', 'absent']
+            )
+        )
+
+        self._client = None
+
+        self.resource_group = None
+        self.name = None
+        self.location = None
+
+        self.sku = None
+        self.size = None
+        self.enable_non_ssl_port = False
+        self.configuration_file_path = None
+        self.shard_count = None
+        self.static_ip = None
+        self.subnet = None
+        self.tenant_settings = None
+
+        self.tags = None
+
+        self.results = dict(
+            changed=False,
+            id=None,
+            host_name=None
+        )
+        self.state = None
+        self.to_do = Actions.NoAction
+
+        self.frameworks = None
+
+        super(AzureRMRedisCaches, self).__init__(derived_arg_spec=self.module_arg_spec,
+                                             supports_check_mode=True,
+                                             supports_tags=True)
+
+    def exec_module(self, **kwargs):
+        """Main module execution method"""
+
+        for key in list(self.module_arg_spec.keys()) + ['tags']:
+            setattr(self, key, kwargs[key])
+
+        old_response = None
+        response = None
+        to_be_updated = False
+
+        # define redis_configuration properties
+        self.redis_configuration_properties = ["maxfragmentationmemory_reserved",
+                                               "maxmemory_reserved",
+                                               "maxmemory_policy",
+                                               "notify_keyspace_events"]
+
+        # get management client
+        self._client = self.get_mgmt_svc_client(RedisManagementClient,
```

Comment:
get client in the module or in the common?

Reply:
it's based on matt/jordans' comments in recent weekly review meeting. to make test efficiency, move client out of azure_rm_common to separate modules.